.github/workflows/build.yml

```diff
@@ -21,12 +21,30 @@ jobs:
         go-arch: ["amd64", "arm", "arm64"]
     steps:
       - uses: actions/checkout@v4
-      - uses: DeterminateSystems/nix-installer-action@main
-      - uses: DeterminateSystems/magic-nix-cache-action@main
       - uses: actions/setup-go@v5
         with:
           go-version: "1.23"
           check-latest: true
+      - name: Cache libraries
+        id: cache-rocksdb
+        if: env.GIT_DIFF
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-rocksdb-lib
+        with:
+          path: /usr/local/lib
+          key: ${{ runner.os }}-rocksdb-${{ matrix.go-arch }}
+      - name: install rocksdb
+        if: env.GIT_DIFF && steps.cache-rocksdb.outputs.cache-hit != 'true'
+        id: install_rocksdb
+        env:
+          ROCKSDB_VERSION: 8.11.3
+        run: |
+          sudo apt update && sudo apt-get install -y zlib1g-dev libbz2-dev libzstd-dev build-essential
+          git clone https://github.com/facebook/rocksdb.git /home/runner/rocksdb
+          cd /home/runner/rocksdb || exit 1
+          git checkout "v$ROCKSDB_VERSION"
+          sudo make -j "$(nproc --all)" shared_lib && sudo ldconfig
```

Comment:
rocksdb is now installed directly on public runners. it is made available on /usr/local/lib